### PR TITLE
First draft of new Python extension template topic

### PR DESCRIPTION
**Pull request by gregvanl**
I've tried to cover the main points of the template without going into too much detail.
@karthiknadig Let me know if you think we need more content on extension authoring best practices or if you've noticed specific questions from authors that need to be addressed here or in the main API docs.

### CODE REVIEW

File: api/advanced-topics/python-extension-template.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+## Adding features from your tool
+
+The generated `bundled/tool/server.py` file is where you will make most of your changes. `TODO` comments in the file point out the various customization points. Also search for `TODO` comments in other locations in the template, such as other Python and Markdown files. You will want to review the LICENSE file, even if you want to keep it MIT License.
+
+## Examples
+
+There are several example implementations created from the template:
+
+* [Pylint](https://github.com/microsoft/vscode-pylint/tree/main/bundled/tool) - implements linting on file `open`, `save`, and `close`.
```
Comment (karthiknadig) [inline api/advanced-topics/python-extension-template.md:64]: We can also include [flake8](https://github.com/microsoft/vscode-flake8/tree/main/bundled/tool) and [Autopep8](https://github.com/microsoft/vscode-autopep8/tree/main/bundled/tool)

File: api/advanced-topics/python-extension-template.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    * `mytool.showNotification`
+1. Following triggers for extension activation:
+    * On Language `python`.
+    * On File with `.py` extension found in the opened workspace.
+    * On Command `mytool.restart`.
+1. Output channel for logging **Output** > **My Tool**.
+
+## Adding features from your tool
+
+The generated `bundled/tool/server.py` file is where you will make most of your changes. `TODO` comments in the file point out the various customization points. Also search for `TODO` comments in other locations in the template, such as other Python and Markdown files. You will want to review the LICENSE file, even if you want to keep it MIT License.
```
[the remark this answers is not in the document]
Comment (gregvanl) [reply api/advanced-topics/python-extension-template.md:58]: I liked calling out the features and TODOs to give the user a preview of what the template provides.
I don't think we need to worry too much about keeping this list in sync with the template as it just gives examples of what the user can expect in the template.
But fine to pull it out if you think it is too much detail

[the remark this answers is not in the document]
Comment (luabud) [reply api/advanced-topics/python-extension-template.md:58]: I like the callout as well, it's good to provide a more concrete example. I'd vote to leave it there!

File: api/advanced-topics/python-extension-template.md (hunk 1 above)
Comment (luabud) [reply api/advanced-topics/python-extension-template.md:64]: Good point! I think it's worth including code actions are also supported.

File: api/advanced-topics/python-extension-template.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (karthiknadig) [reply api/advanced-topics/python-extension-template.md:58]: Let us leave it here.

File: api/advanced-topics/python-extension-template.md (hunk 1 above)
Comment (gregvanl) [reply api/advanced-topics/python-extension-template.md:64]: @karthiknadig Is the autopep8 formatter extension available on the Marketplace? I can find the others but not that one? Maybe it is still under development?

Comment (gregvanl) [reply api/advanced-topics/python-extension-template.md]: Added new extensions in last commit as well as call out Code Actions

**Comment by gregvanl** [comment]
Merging after a side conversation with @luabud. We can tune more after reviewing on the staging server

**Pull request merged**